action/rlp_tx.go

```diff
@@ -0,0 +1,76 @@
+package action
+
+import (
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/rlp"
+	"github.com/iotexproject/go-pkgs/hash"
+	"github.com/iotexproject/iotex-address/address"
+	"github.com/pkg/errors"
+	"golang.org/x/crypto/sha3"
+
+	"github.com/iotexproject/iotex-core/config"
+)
+
+func init() {
+	// register the extern chain ID for test
+	config.SetExternChainID(config.Default.Chain.ExternChainID)
+}
+
+type rlpTransaction interface {
+	Nonce() uint64
+	GasPrice() *big.Int
+	GasLimit() uint64
+	Recipient() string
+	Amount() *big.Int
+	Payload() []byte
+}
+
+func rlpRawHash(tx rlpTransaction, chainID uint32) hash.Hash256 {
+	rawTx, err := generateRlpTx(tx)
+	if err != nil {
+		return hash.ZeroHash256
```

Comment:
return error and leave the handling of the error to where it is called

Reply:
since the caller is sealed.Hash(), it does not return err, so return all-0 hash to indicate error

added a check to catch all-0 hash in Sing() and Verify()

Reply:
even this is a private function, it is still a good practice to return the error and let the caller handle the error properly.